src/grpc/zero_copy_stream.cc

```diff
@@ -65,9 +68,40 @@ bool GrpcZeroCopyInputStream::Next(const void** data, int* size) {
 void GrpcZeroCopyInputStream::BackUp(int count) {
   if (0 < count && static_cast<size_t>(count) <= position_) {
     position_ -= count;
+    bytes_read_ -= count;
   }
 }
 
+bool GrpcZeroCopyInputStream::Skip(int count) {
+  if (count < 0) {
+    // Safe guard against wrong usage.
+    return false;
+  }
+  size_t count_left = static_cast<size_t>(count);
+  while (position_ + count_left > current_buffer_size_) {
+    // Skipping past the current buffer, read the next one.
+    int delta = static_cast<int>(current_buffer_size_ - position_);
+    count_left -= delta;
+    bytes_read_ += delta;
+    position_ = 0;
```

Comment:
Are you sure setting the position to 0? I assumed `Next` should be called from the caller function.

```
In the end-of-stream case, the stream is advanced to the end of the stream.
```

Reply:
serrialize_.Next() will set current_bufer_size_ to 0,   with position_=0. it is end of the stream.   What is the problem?

Reply:
Ack, I understand now.